numbers/randBeta.go

```diff
@@ -0,0 +1,182 @@
+package numbers
+
+import (
+	"math"
+	"math/rand"
+	"log"
+)
+
+const RelError float64 = 1.0e-8
+const Small float64 = 1.0e-30
+const MaxIterations int = 200
+
+//RandBeta returns a random number drawn from a beta distribution with parameters alpha and beta as an xy point where y is the function density.
+//Implemented from principles in R. C. H. Cheng (1978). Generating beta variates with nonintegral shape parameters. Communications of the ACM 21, 317-322.
+func RandBeta(alpha float64, beta float64) float64 {
+	r := rand.Float64()
+	for r <= 0.0 || r >= 1.0 {//catches the case where r is exactly  0.0 or 1.0
```

Comment:
I am not sure we need r as a random float, but rand.Float64() is left closed right open, so I don't think it will ever be 1